### ISSUE TITLE
remove "cant find escaltion tool if user is already root

### ISSUE THREAD

**Issue by solomoncyj** [ChrisTitusTech/linutil#816]
## Describe the bug
log in as root user without a escation tool, eg docker

## To Reproduce
run the script
select an opetion which requires elevation
## Expected behavior
the script to run normal

## Checklist
- [*] I checked for duplicate issues.
- [*] I checked already existing discussions.
- [*] This issue is not included in the roadmap.
- [*] This issue is present on both stable and development branches.

**Comment by solomoncyj** [comment]
![image](https://github.com/user-attachments/assets/812d0472-d198-45b3-842d-4ca0289fb38b)

**Comment by jeevithakannan2** [comment]
Linutil should not be ran as root. Will add restriction to prevent running as root.

**Comment by solomoncyj** [comment]
> Linutil should not be ran as root. Will add restriction to prevent running as root.

how should i make my docker container **not** log me in as root?

**Comment by jeevithakannan2** [comment]
You can specify a new user in the docker file and run it with `docker run --user myuser myimage`. If you're using an image from the docker hub most of them have a non-root user by default.

**Comment by cartercanedy** [comment]
I mean, we could make this work with a superuser check and setting `ESCALATION_TOOL=""` in `common_script.sh` if run with a root session, right?

**Comment by cartercanedy** [comment]
There might be some other issues if relative dirs are used in any of the scripts, but that's generally discouraged anyways

**Comment by lj3954** [comment]
> I mean, we could make this work with a superuser check and setting `ESCALATION_TOOL=""` in `common_script.sh` if run with a root session, right?

All scripts would also need to be updated to not quote the escalation tool variable in that case, since otherwise they'd be attempting launch a binary with an empty name in your PATH, which obviously cannot exist.

**Comment by cartercanedy** [comment]
That could get quickly solved by redirecting to a custom escalation function that can implement the branching logic to check for the root user case. Would be as simple as implementing the fn in `common-script.sh` and a sed replace for `"$ESCALATION_TOOL"`. I'd do it myself, but I'm not at home, atm

**Comment by jeevithakannan2** [comment]
```sh
elevated_execution() {
    if [ -z "$ESCALATION_TOOL" ]; then
        "$@"
    else
        "$ESCALATION_TOOL" "$@"
    fi
}
```

Need to replace every `ESCALATION_TOOL` in scripts with `elevated_execution` function

Is this simple function good in `common-script.sh` ??

**Comment by cartercanedy** [comment]
Nice! Yeah, that's exactly what I'm thinking. I'd probably use a `find ./ ! -name 'common-script.sh' -type f -name '*.sh'` to iterate over all of the scripts and run `sed -i'' 's/"$ESCALATION_TOOL"/elevated_execution/g'

**Comment by jeevithakannan2** [comment]
Yeah was thinking of same idea

**Comment by lj3954** [comment]
> ```shell
> elevated_execution() {
>     if [ -z "$ESCALATION_TOOL" ]; then
>         "$@"
>     else
>         "$ESCALATION_TOOL" "$@"
>     fi
> }
> ```
> 
> Need to replace every `ESCALATION_TOOL` in scripts with `elevated_execution` function
> 
> Is this simple function good in `common-script.sh` ??

You'll need to remove the quotes from `$@`, quoting it will cause the shell to treat the entire list of arguments as one command.

**Comment by cartercanedy** [comment]
> You'll need to remove the quotes from `$@`, quoting it will cause the shell to treat the entire list of arguments as one command.

Quoting `$@` should still work. I'm fairly certain that word expansion of quoted `$@` is posix, same with the non-expansion of quoted`$*`

**Comment by lj3954** [comment]
Fixed in #927
